paf-mvp-core-js/src/validation/io.validator.ts

```diff
@@ -0,0 +1,35 @@
+import { MessageVerificationResult } from '@core/crypto';
+import { IdsPrefsResponse } from '@core/model';
+import { getTimeStampInSec } from '@core/timestamp';
+import { MessageConsistencyValidator } from './io-consistency.validator';
+import { IIO_DSAValidator } from './io-signature.validator';
+
+export class IdsPrefsResponseValidator {
+  constructor(
+    private ioDsaVerifier: IIO_DSAValidator,
+    private consistencyVerifier = new MessageConsistencyValidator()
+  ) {}
+
+  async verifySignatureAndContent(
+    request: IdsPrefsResponse,
+    senderHost: string,
+    receiverHost: string,
+    timestampInSec: number = getTimeStampInSec()
```

Comment:
With a richer model, it would simplify a lot:
```typescript
//////////////////////////////
// Module: Requests and Responses
/////////////////////////////

export interface AuthenticatedResponse { // Consider a class instead
    /**
     * Domain of the sender described in the response.
     */
    sender: Domain;
    /**
     * Domain of the receiver described in the response.
     */
    receiver: Domain;
    /**
     * Timestamps generated by the receiver.
     */
    timestamp: Timestamp;
    signature: Signature;
    /**
     * Can be implemented as a generated property by the model (class) or set via a service.
     */
    signedString: string;

    metaData: ResponseMedata;
}

export interface ResponseMedata {
  /**
   * Timestamp of witch the Response has been received.
   * Usefull to verify some TTL
   */
  receptionTimestamp: Timestamp;
  /**
   * Domain of the sender from the Receiver perspective.
   */
  senderDomain: Domain;
  /**
   * Domain of the receiver from the Receiver perspective.
   */
  receiverDomain: Domain;
}

export interface GetIdsPrefsResponse extends AuthenticatedResponse {
  body: IdsAndOptionalPreferences;
}

// Alternative a bit more complexe - Not sure if it worth it

export interface AuthenticatedResponse2<Content> {
    sender: Domain;
    receiver: Domain;
    timestamp: Timestamp;
    signature: Signature;
    signedString: string; // Can be implemented as a generated property by the model (class) or set via a service.

    metaData: ResponseMedata;
    content: Content,
}

export interface GetIdsPrefsResponse extends AuthenticatedResponse2<IdsAndOptionalPreferences> {}

//////////////////////////////
// Module: Generic validation module
/////////////////////////////

interface Validation {
    success: boolean,
    errors?: Error[],
}

interface Validator<M> {
    validate(model: M): Promise<Validation>;
}

class ChainValidator<M> implements Validator<M> {
    constructor(
        private validators: Validator<M>[],
    ) {}

    validate(model: M): Promise<Validation> {
        throw 'To implement: iterate over all validator. Stop if error';
    }
}
 

//////////////////////////////
// Module: AuthenticatedResponse Validator
/////////////////////////////

export class AuthenticatedResponseConsistencyValidator implements Validator<AuthenticatedResponse> {
    constructor(public messageTTLinSec = 30) {}
  
    validate(response: AuthenticatedResponse):  Promise<Validation> {
        throw 'To implement: check with meta data';
    }
}

export class AuthenticatedResponseDSAValidator implements Validator<AuthenticatedResponse> {
    // ...

    validate(response: AuthenticatedResponse): Promise<Validation> {
        throw 'To implement';
    }
}

//////////////////////
/// Usage
/////////////////////

export const AuthenticatedResponseValidator = new ChainValidator<AuthenticatedResponse>([
    new AuthenticatedResponseConsistencyValidator(/* ... */),
    new AuthenticatedResponseDSAValidator(/* ... */),
]);

```